src/imagewriter.cpp

```diff
@@ -342,10 +342,49 @@ void ImageWriter::startWrite()
     if (_multipleFilesInZip)
     {
         static_cast<DownloadExtractThread *>(_thread)->enableMultipleFileExtraction();
-        DriveFormatThread *dft = new DriveFormatThread(_dst.toLatin1(), this);
-        connect(dft, SIGNAL(success()), _thread, SLOT(start()));
-        connect(dft, SIGNAL(error(QString)), SLOT(onError(QString)));
-        dft->start();
+        if(_initFormat != "UNRAID") {
```

Comment:
I'm a bit concerned about this code as we had a bunch of issues with strange formatting behavior on the old USB creator. Do you see any potential for issues with the download extraction thread happening here, or the boot images having issues? If not it's probably okay, it's just that the old raspberry pi one was so much simpler - I'm almost wondering if it's possible to do this rename with that (at least for non-legacy boot)

Reply:
well, `_initFormat` is something that gets read from the os list json, so the `if` part of this is mainly here to preserve the rpi imager's original behavior in the event that the user chooses to flash a custom img file, since we've left that option in there. the else portion will be what executes when we're actually dealing with a zip downloaded from the unraid os list, and it's already intended to mimic the behavior of the original usb creator (pulled from this section here https://github.com/unraid/usb-creator/blob/65f7b9d5218a55dd7d6df7edc62c174bcb664efc/creator.cpp#L1251 ).

from what i can tell in the original usb creator, it seems to go through 3 steps regardless of boot selection - first it flashes the selected boot img file, then it extracts the unraid zip contents, then it runs syslinux. to see how rpi imager would handle flashing the boot img file independently first, i had selected one of those from my local clone of usb-creator, and noticed that, despite it being a local file, `DownloadExtractThread` was what rpi would select on its own to flash the .img.gz file, so that's what's being done manually on line 376 here to flash the .img.gz first (`imgThread`). then the signal for that thread's completion is connected to the start of the regular `DownloadExtractThread` (`_thread`), so once the .img.gz file has been flashed, it immediately kicks off the extraction of the files from the unraid zip.

does that address your concerns, or did i misunderstand what you were asking about?

Reply:
Let's chat more about this later today - I put the comment above on the wrong line IMO, I think it's smart to preserve the functionality of the Raspi Creator, I'm just wondering if there isn't a more elegant way to write the drive than reading sector sizes using magic strings.